### ISSUE TITLE
Apply scrollback reflow patch ?

### ISSUE THREAD

**Issue by ShellCode33** [LukeSmithxyz/st#345]
Hey, thanks for your work ! Love the custom dmenu stuff. Would you mind applying the [scrollback reflow patch](https://st.suckless.org/patches/scrollback/st-scrollback-reflow-0.8.5.diff) ?

> Apply the following patch on top of the [original scrollback patch (=>0.8.5)](https://st.suckless.org/patches/scrollback/st-scrollback-0.8.5.diff) to allow column and row reflow.

Very useful when you resize windows or switch to fullscreen.

Take care

**Comment by LukeSmithxyz** [comment]
How is this any different from the current functionality?

**Comment by LukeSmithxyz** [comment]
Okay. Yeah I'll try to add it, but looks like there are a bunch of little incompatibilities with other patches I or someone else will have to figure our first. I'll get to it if I can.

**Comment by ShellCode33** [comment]
Yep, many incompatibilities indeed. At first I was willing to submit a PR but I gave up when I saw the amount of lines in the rejected diff. Ended up doing my own fork of st, which I'm pretty sure saved me some time. Even for you I wonder if it wouldn't be faster to reapply all your patches from scratch using a clean base of st 9.0. Many of the fixes you have here on this repo are probably also fixed upstream and some others are now available as patches (such as [glyph_wide_support](https://st.suckless.org/patches/glyph_wide_support/)).

Anyway, good luck to anyone reading this and willing to put in the effort

**Comment by LukeSmithxyz** [comment]
The main issue is probably getting this, boxdraw and ligatures all to work in one place. I'll try to repatch from scratch, but I think it will have to grapple with this same problem.

**Comment by ShellCode33** [comment]
I feel like the boxdraw patch is not worth it if you're already using a "good" font. I use Fantasque Sans Mono for my terminal, and it looks pretty much the same with and without boxdraw. At least the difference is not noticeable enough to me to add boxdraw. I added NotoSansMono as comparison, it doesn't render well indeed. It's up to you of course but for me I choose to rely on Fantasque Sans Mono instead of messing up my code even more using boxdraw. Maybe you should consider getting rid of it.

NotoSansMono font on my fork of st, without boxdraw (pretty ugly I agree) :

![image](https://user-images.githubusercontent.com/8455652/202047960-9cb3c3c7-5236-4363-81d8-e1e333cd1b2b.png)

NotoSansMono font on your fork of st, with boxdraw :

![image](https://user-images.githubusercontent.com/8455652/202048127-5a7ee792-a9fb-4597-af8c-e355ff15e2d6.png)

Fantasque Sans Mono on my fork of st, without boxdraw :

![image](https://user-images.githubusercontent.com/8455652/202048700-945fbdf8-68a5-42d3-9326-96a3b0fcda06.png)

Fantasque Sans Mono on your fork of st, with boxdraw :

![image](https://user-images.githubusercontent.com/8455652/202048594-6bfa1058-5e47-49af-af97-83b7634a98ac.png)

**Comment by LukeSmithxyz** [comment]
Yeah it's not that bad. I had added boxdraw from public outcry, but if it's either or, I might go with this instead due to the wider functionality.

**Comment by speediegq** [comment]
The original redraw patch that was mentioned was suggested in the st-flexipatch repository but never added because it is so massive and conflicts with a lot of patches.

There is a much smaller patch that achieves nearly the same functionality. See [this patch](https://raw.githubusercontent.com/nimaipatel/st/master/patches/7672445bab01cb4e861651dc540566ac22e25812.diff). I have this in my own build and it works great!

Only functionality this version seems to miss is reflow for the top, but text does seem to redraw for example when you're changing mfact in dwm.

This version written by nimaipatel does not include scrollback but it can be applied on top ofl scrollback without issues!

**Comment by iStagnant** [comment]
@LukeSmithxyz if you have time and interested in repatching st using version 0.9 as base you can see my [st build](https://github.com/iStagnant/st) it has these patches applied:
![image](https://user-images.githubusercontent.com/82290573/218145918-a3c3e437-7dd6-432f-b910-77f95aefac10.png)
it has the patch suggested in this issue and glyph wide support from [my PR](https://github.com/LukeSmithxyz/st/pull/349) and the ligature patch that i used is from your [original commit](https://github.com/LukeSmithxyz/st/commit/4edc7282ab841142e970dcb4f03f251bb5ed7292).
